support/modules/file-install

```diff
@@ -21,19 +23,35 @@ case "$STATE" in
 
     ArtifactInstall)
         dest_dir=$(cat $dest_dir_file)
-        [[ -d $dest_dir ]] || install -d $dest_dir
-        tar -cf ${prev_files_tar} -C ${dest_dir} .
-        rm -rf ${dest_dir}
-        mkdir -p ${dest_dir}
+        single_file=$(cat $single_file_file)
+        test -z "$dest_dir" -o -z "$single_file" && exit 1
+        mkdir -p $dest_dir
+        if [ $single_file == "true" ]; then
+            mkdir -p $tmp_dest_dir
+            filename=$(tar -tf ${update_files_tar})
+            [[ -f ${dest_dir}/${filename} ]] && tar -cf ${prev_files_tar} -C ${dest_dir} ${filename}
+        else
+            test "$dest_dir" == "/" && exit 1
```

Comment:
This test should probably done as early as possible, no point in doing anything if above condition is true. Do not know if it is valid to have at the top outside of the states?

Reply:
I still want to support deploying single files to `/`, so the check needs to be done here.

Reply:
Also, when I originally wrote the Update Modules I had these statements out of the states.

What happens is that for some states the files are not present. So if we want to execute the script with -e (exit on errors), we cannot have the checks in the top level context.

Reply:
Thanks for explaining